api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -703,6 +710,27 @@ private Boolean copyBlob(String bucketName, Blob b) {
       }
     }
 
+    // If a lock is held by another user, log this to establish a rough estimate of how often
+    // locked notebooks are encountered. Note that this only covers locks encountered from the
+    // Workbench - any Jupyter UI-based lock detection does not touch this code path.
+    String currentUsername = userProvider.get().getUsername();
+    boolean notebookAvailable =
+        response.getLockExpirationTime() == null
+            || response.getLastLockedBy() == null
+            || response.getLockExpirationTime() < clock.millis()
+            || response.getLastLockedBy().equals(currentUsername);
+    if (notebookAvailable) {
+      log.info(
+          String.format(
+              "user '%s' observed notebook locked by '%s'",
+              currentUsername, response.getLastLockedBy()));
+    }
+    monitoringService.recordBundle(
+        MeasurementBundle.builder()
+            .addEvent(GaugeMetric.NOTEBOOK_LOCK_CHECK)
+            .addTag(MetricLabel.NOTEBOOK_AVAILABLE_TO_USER, Boolean.toString(notebookAvailable))
```

Comment:
Just a heads up: I'm about 75% sure that Stackdriver considers the Labels (tags) list to be part of the metric descriptor, so if you want to add another tag in the future, we have to either drop this one and replace it, or add a new one. I'll put this in the docs.